app/controllers/admin/users_controller.rb

```diff
@@ -169,5 +194,21 @@ def sort_direction
     def set_user
       @user = user_scope.find params[:id].to_i
     end
+
+    def handle_search_query
+      return unless params[:q].present?
+
+      search_params = ActionController::Parameters.new({ q: params[:q] })
+      permitted_params = ClientSearchQuery.permit_params(search_params)
+      return unless permitted_params.present?
+
+      @search_query = ClientSearchQuery.find_or_create_by_params(permitted_params, user: current_user)
+      return if @search_query.errors.any?
+
+      # Redirect to the search query URL if this is a GET request
+      redirect_to admin_user_search_query_path(@search_query) if request.get?
```

Comment:
maybe this fixes the brakeman warning
```suggestion
      redirect_to admin_user_search_query_path(@search_query) if request.get? || request.head?
```